New Spawns: StrawWagen's Placements

TechnichalDoge opened the pull request.

TechnichalDoge committed:
Insert Straw's idea

TechnichalDoge commented:
[https://cdn.discordapp.com/attachments/347623692823560193/729839544324718672/20200706162243_1.jpg](url)

TechnichalDoge commented:
[https://cdn.discordapp.com/attachments/574486617155436544/729846090542874665/20200706164850_1.jpg](url)

TechnichalDoge committed:
Update sv_config.lua

overlordbleck commented:
What exactly is different here?

TechnichalDoge commented:
New spawnpoints

TechnichalDoge commented:
![image](https://user-images.githubusercontent.com/16617746/86679594-555e6900-bfc3-11ea-8609-133888a36b6d.png)

TechnichalDoge commented:
courtesy of strawwagen

TechnichalDoge commented:
![at_the_20_hour](https://user-images.githubusercontent.com/16617746/86679758-7d4dcc80-bfc3-11ea-90d7-a045df5a8760.png)

StrawWagen commented:
replying to @overlordbleck , I redid the spawns for the map, (roughly) basing their placement off of the spawns on bigcity.

I transferred the outline of bigcity's spawn (represented with ropes in the screenshot logica posted) to a dupe and tried to fit it in an enclosed part of novenka, I decided that in the buildings would be good, and placed a continuous ring of spawns that outline an area very roughly equal to the area of bigcity's spawn.

I placed each spawnpoint next to a wall, so anyone who spawns at them will have a flank always covered.

I decided to use the footprint of bigcity's spawn since that map has really quick engagement times, and I feel that the current spawnpoints are too far apart.

TechnichalDoge commented:
this seems to be the more discussion orientated one, so lets discuss this PR https://github.com/CFC-Servers/cfc_random_spawn/pull/13 with this discussion

TechnichalDoge commented:
I chose the spawnpoints i did simularly, but not really to Straw. I tried to pick ones that would best fit if it was like bigciy, which has a big open space encircled by spawns and alleys. Instead of using a bigcity outline, i just expanded it to what seemed fitting to novenka

TechnichalDoge commented:
i also think that the debate between these two should be voted on in announcements or something

StrawWagen commented:
![20200706174810_1](https://user-images.githubusercontent.com/64710817/86693189-4ecdf280-bfbf-11ea-9ef3-797bafae4565.jpg)
![20200706174815_1](https://user-images.githubusercontent.com/64710817/86693193-4f668900-bfbf-11ea-810c-b9037a2ff69d.jpg)

Here's the spawn locations from different angles, all the yellow poles are them.

TechnichalDoge commented:
@StrawWagen could you post images of my version, as well?

StrawWagen commented:
![20200706180218_1](https://user-images.githubusercontent.com/64710817/86701213-acb20880-bfc6-11ea-85be-2f1a3d56b465.jpg)
![20200706180233_1](https://user-images.githubusercontent.com/64710817/86701216-ad4a9f00-bfc6-11ea-86c9-beb5fe4af3a8.jpg)
heeere they are @TechnichalDoge

ghost commented:
Is this supposed to be merged with https://github.com/CFC-Servers/cfc_random_spawn/pull/13 or is this a choice between the two?

TechnichalDoge committed:
Update sv_config.lua

StrawWagen commented:
the latest update has changes suggested by decline, as well it includes angles and pvp states of the spawns

TechnichalDoge committed:
Update sv_config.lua

STRAWWWWWWWWWWWWWWWWWAAAAAAAAAAAAAGGGGGGGGGGGGGGGGEEEEEEEEEENNNNNNNNNNNNNNNNNNNNNNNNNNNNNN

TechnichalDoge commented:
@Periapsises #13 was an alternative, the ones i suggested, so a choice, but people overwelmingly voted it down when posted with straw's in sugdisc

plally commented on lua/cfc_random_spawn/sv_config.lua:
```diff
@@ -22,45 +22,68 @@ CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_bluehills_test3"] = {
-    { spawnPos = Vector( 3910, 7229, -184 ) }, -- Angle( 0, 135, 0 )
-    { spawnPos = Vector( 3872, 6256, -240 ) }, -- Angle( 0, 90, 0 )
-    { spawnPos = Vector( 2990, 8304, -264 ) } -- Angle( 0, 135, 0 )
+-- CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_novenka"] = { --original
```
we can always look through the github history if we need the old spawnpoints, commenting them probably isnt necessary

TechnichalDoge replied:
fair enough

TechnichalDoge committed:
Update sv_config.lua

plally reviewed (commented):
this is fine , although im not sure the keys spawnAngle and pvp actually do anything. perhaps someone could provide more info as to why they were added?

it looks like the option to have pvp only spawns was removed a few commits back?

there could be value in keeping the pvp key if pvp only spawns are to be added back in the future

StrawWagen commented:
@PLally phatso gave me a table to fill in, i filled it in.
https://discordapp.com/channels/225975453138026497/390548646233636865/730509597969743872

I wondered the same thing, but phatso's the person who has the answer to your question, it's probably for future-proofing

sarahsturgeon commented:
We eventually want to also include the spawn angle. should be simple enough, just wanted to support it before the code was written

Deconkle commented:
this code does not error

StrawWagen commented:
looks like it's ready to merge then

TechnichalDoge committed:
Merge branch 'master' into master

TechnichalDoge commented:
@Deconkle

TechnichalDoge commented:
@overlordbleck @brandonsturgeon whoever gets here first can get the honor of merging! :smile:

overlordbleck commented on lua/cfc_random_spawn/sv_config.lua:
```diff
@@ -22,42 +22,25 @@ CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_bluehills_test3"] = {
-    { spawnPos = Vector( 3910, 7229, -184 ) }, -- Angle( 0, 135, 0 )
-    { spawnPos = Vector( 3872, 6256, -240 ) }, -- Angle( 0, 90, 0 )
-    { spawnPos = Vector( 2990, 8304, -264 ) } -- Angle( 0, 135, 0 )
+CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_novenka"] = { -- strawwagm_venka
```
```suggestion
CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_novenka"] = {
```

overlordbleck commented on lua/cfc_random_spawn/sv_config.lua:
```diff
@@ -22,42 +22,25 @@ CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_bluehills_test3"] = {
-    { spawnPos = Vector( 3872, 6256, -240 ) }, -- Angle( 0, 90, 0 )
-    { spawnPos = Vector( 2990, 8304, -264 ) } -- Angle( 0, 135, 0 )
+CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_novenka"] = { -- strawwagm_venka
+    { spawnPos = Vector( 6052, 11677, -383 ), spawnAngle = Angle( 0, 0, -135 ), pvp = true },
```
Shouldn't the spawn angle be in the yaw and not the roll...? Unless we want people to spawn sideways I guess...

TechnichalDoge replied:
@brandonsturgeon @StrawWagen

TechnichalDoge committed:
Update lua/cfc_random_spawn/sv_config.lua

Co-authored-by: OverlordBleck <40676794+overlordbleck@users.noreply.github.com>

TechnichalDoge committed:
Update sv_config.lua

StrawWagen commented on lua/cfc_random_spawn/sv_config.lua:
```diff
@@ -23,41 +23,24 @@ CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_bluehills_test3"] = {
-    { spawnPos = Vector( 3910, 7229, -184 ) }, -- Angle( 0, 135, 0 )
-    { spawnPos = Vector( 3872, 6256, -240 ) }, -- Angle( 0, 90, 0 )
-    { spawnPos = Vector( 2990, 8304, -264 ) } -- Angle( 0, 135, 0 )
+    { spawnPos = Vector( 6052, 11677, -383 ), spawnAngle = Angle( 0, -135, 0 ), pvp = true },
+    { spawnPos = Vector( 6364, 10733, -383 ), spawnAngle = Angle( 0, -90, 0 ), pvp = true },
+    { spawnPos = Vector( 9024, 10741, -383 ), spawnAngle = Angle( 0, -148, 0 ), pvp = true },
+    { spawnPos = Vector( 5334, 11687, -383 ), spawnAngle = Angle( 0, -45, 0 ), pvp = true },
+    { spawnPos = Vector( 4884, 11390, -383 ), spawnAngle = Angle( 0, -45, 0 ), pvp = true },
+    { spawnPos = Vector( 4875, 10749, -383 ), spawnAngle = Angle( 0, 42, 0 ), pvp = true },
+    { spawnPos = Vector( 6417, 5878, -383 ), spawnAngle = Angle( 0, 135, 0 ), pvp = true },
+    { spawnPos = Vector( 8995, 6627, -383 ), spawnAngle = Angle( 0, 135, 0 ), pvp = true },
+    { spawnPos = Vector( 8984, 7792, -383 ), spawnAngle = Angle( 0, -180, 0 ), pvp = true },
+    { spawnPos = Vector( 8989, 9103, -383 ), spawnAngle = Angle( 0, -135, 0 ), pvp = true },
+    { spawnPos = Vector( 9043, 9799, -383 ), spawnAngle = Angle( 0, 124, 0 ), pvp = true },
+    { spawnPos = Vector( 7648, 10733, -383 ), spawnAngle = Angle( 0, -90, 0 ), pvp = true },
+    { spawnPos = Vector( 8486, 5880, -383 ), spawnAngle = Angle( 0, 155, 0 ), pvp = true },
+    { spawnPos = Vector( 7265, 5878, -383 ), spawnAngle = Angle( 0, 45, 0 ), pvp = true },
+    { spawnPos = Vector( 5578, 6911, -383 ), spawnAngle = Angle( 0, 45, 0 ), pvp = true },
+    { spawnPos = Vector( 5121, 8820, -127 ), spawnAngle = Angle( 0, 90, 0 ), pvp = true },
+    { spawnPos = Vector( 4475, 8824, -127 ), spawnAngle = Angle( 0, 59, 0 ), pvp = true },
+    { spawnPos = Vector( 4497, 9553, -127 ), spawnAngle = Angle( 0, 0, 8 ), pvp = true },
+    { spawnPos = Vector( 5654, 8670, -383 ), spawnAngle = Angle( 0, 0, 0 ), pvp = true },
+    { spawnPos = Vector( 5694, 7802, -383 ), spawnAngle = Angle( 0, 0, 0 ), pvp = true }
```
```suggestion
    { spawnPos = Vector( 8989, 9103, -383 ), spawnAngle = Angle( 0, -135, 0 ), pvp = true },	
    { spawnPos = Vector( 9043, 9799, -383 ), spawnAngle = Angle( 0, 124, 0 ), pvp = true },	
    { spawnPos = Vector( 8984, 7792, -383 ), spawnAngle = Angle( 0, -180, 0 ), pvp = true },	
    { spawnPos = Vector( 9024, 10741, -383 ), spawnAngle = Angle( 0, -148, 0 ), pvp = true },	
    { spawnPos = Vector( 7648, 10733, -383 ), spawnAngle = Angle( 0, -90, 0 ), pvp = true },	
    { spawnPos = Vector( 8995, 6627, -383 ), spawnAngle = Angle( 0, 135, 0 ), pvp = true },	
    { spawnPos = Vector( 6364, 10733, -383 ), spawnAngle = Angle( 0, -90, 0 ), pvp = true },	
    { spawnPos = Vector( 8486, 5880, -383 ), spawnAngle = Angle( 0, 155, 0 ), pvp = true },	
    { spawnPos = Vector( 5654, 8670, -383 ), spawnAngle = Angle( 0, 0, 0 ), pvp = true },	
    { spawnPos = Vector( 5694, 7802, -383 ), spawnAngle = Angle( 0, 0, 0 ), pvp = true },	
    { spawnPos = Vector( 7265, 5878, -383 ), spawnAngle = Angle( 0, 45, 0 ), pvp = true },	
    { spawnPos = Vector( 6052, 11677, -383 ), spawnAngle = Angle( 0, -135, 0 ), pvp = true },	
    { spawnPos = Vector( 5121, 8820, -127 ), spawnAngle = Angle( 0, 90, 0 ), pvp = true },	
    { spawnPos = Vector( 5578, 6911, -383 ), spawnAngle = Angle( 0, 45, 0 ), pvp = true },	
    { spawnPos = Vector( 6417, 5878, -383 ), spawnAngle = Angle( 0, 135, 0 ), pvp = true },	
    { spawnPos = Vector( 4875, 10749, -383 ), spawnAngle = Angle( 0, 42, 0 ), pvp = true },	
    { spawnPos = Vector( 5334, 11687, -383 ), spawnAngle = Angle( 0, -45, 0 ), pvp = true },	
    { spawnPos = Vector( 4497, 9553, -127 ), spawnAngle = Angle( 0, 8, 0 ), pvp = true },	
    { spawnPos = Vector( 4475, 8824, -127 ), spawnAngle = Angle( 0, 59, 0 ), pvp = true },	
    { spawnPos = Vector( 4884, 11390, -383 ), spawnAngle = Angle( 0, -45, 0 ), pvp = true },	
```

TechnichalDoge replied:
wait, wtf is this do

StrawWagen replied:
i did the angle fix through code

StrawWagen commented on lua/cfc_random_spawn/sv_config.lua:
```diff
@@ -23,41 +23,24 @@ CFCRandomSpawn.Config.CUSTOM_SPAWNS["gm_bluehills_test3"] = {
+    { spawnPos = Vector( 5578, 6911, -383 ), spawnAngle = Angle( 0, 45, 0 ), pvp = true },
+    { spawnPos = Vector( 5121, 8820, -127 ), spawnAngle = Angle( 0, 90, 0 ), pvp = true },
+    { spawnPos = Vector( 4475, 8824, -127 ), spawnAngle = Angle( 0, 59, 0 ), pvp = true },
+    { spawnPos = Vector( 4497, 9553, -127 ), spawnAngle = Angle( 0, 0, 8 ), pvp = true },
```
```suggestion
    { spawnPos = Vector( 4497, 9553, -127 ), spawnAngle = Angle( 0, 8, 0 ), pvp = true },
```
Missed angle fix

TechnichalDoge committed:
Update lua/cfc_random_spawn/sv_config.lua

Co-authored-by: StrawWagen <64710817+StrawWagen@users.noreply.github.com>

sarahsturgeon reviewed (approved):
shore, why not

sarahsturgeon commented:
ok to test

TechnichalDoge commented:
"shore, why not"
if your drunk log out of github

StrawWagen commented:
??????? why man
